be/src/exec/avro_scanner.cpp

```diff
@@ -0,0 +1,525 @@
+        }
+        _cur_file_reader = nullptr;
+    }
+}
+
+Status AvroScanner::open_file_reader() {
+    if (_cur_file_reader != nullptr) {
+        if (_stream_load_pipe != nullptr) {
+            _stream_load_pipe.reset();
+            _cur_file_reader = nullptr;
+        } else {
+            delete _cur_file_reader;
+            _cur_file_reader = nullptr;
+        }
+    }
+
+    const TBrokerRangeDesc& range = _ranges[_next_range];
+
+    switch (range.file_type) {
+    case TFileType::FILE_STREAM: {
+        _stream_load_pipe = _state->exec_env()->load_stream_mgr()->get(range.load_id);
+        if (_stream_load_pipe == nullptr) {
+            VLOG_NOTICE << "unknown stream load id: " << UniqueId(range.load_id);
+            return Status::InternalError("unknown stream load id");
+        }
+        _cur_file_reader = _stream_load_pipe.get();
+        break;
+    }
+    case TFileType::FILE_LOCAL:
+    case TFileType::FILE_BROKER:
+    case TFileType::FILE_S3:
+    default: {
+        std::stringstream ss;
+        ss << "Only support stream type. Unsupport file type, type=" << range.file_type;
+        return Status::InternalError(ss.str());
+    }
+    }
+    _cur_reader_eof = false;
+    return Status::OK();
+}
+
+Status AvroScanner::open_avro_reader() {
+    if (_cur_avro_reader != nullptr) {
+        delete _cur_avro_reader;
+        _cur_avro_reader = nullptr;
+    }
+
+    _cur_avro_reader = new AvroReader(_state, _counter, _profile, _cur_file_reader, nullptr);
+    RETURN_IF_ERROR(_cur_avro_reader->init());
+    return Status::OK();
+}
+
+Status AvroScanner::open_next_reader() {
+    if (_next_range >= _ranges.size()) {
+        _scanner_eof = true;
+        return Status::OK();
+    }
+
+    RETURN_IF_ERROR(open_file_reader());
+    RETURN_IF_ERROR(open_avro_reader());
+    _next_range++;
+
+    return Status::OK();
+}
+
+////// class AvroReader
+AvroReader::AvroReader(RuntimeState* state, ScannerCounter* counter, RuntimeProfile* profile,
+                       FileReader* file_reader, LineReader* line_reader)
+        : _next_line(0),
+          _total_lines(0),
+          _state(state),
+          _counter(counter),
+          _profile(profile),
+          _file_reader(file_reader),
+          _line_reader(line_reader),
+          _closed(false),
+          _decoder(avro::binaryDecoder()),
+          _in(nullptr) {
+    _bytes_read_counter = ADD_COUNTER(_profile, "BytesRead", TUnit::BYTES);
+    _read_timer = ADD_TIMER(_profile, "ReadTime");
+    _file_read_timer = ADD_TIMER(_profile, "FileReadTime");
+}
+
+AvroReader::~AvroReader() {
+    _close();
+}
+
+Status AvroReader::init() {
+    bool exist = FileUtils::check_exist(config::avro_schema_file_path);
+    if (!exist) {
+        return Status::InternalError("there is no avro schema file at " +
+                                     config::avro_schema_file_path +
+                                     ". Please put an schema file in json format.");
+    } else {
```

Comment:
The else statement does not appear to be required.
```suggestion
    } 
```